### PR TITLE
Fix: ht bucket index calculation

### PR DESCRIPTION
**Pull request by matthiasbeyer**

### CODE REVIEW

**Comment by matthiasbeyer** [comment]
Waiting on #228

File: src/libreset/ht/base.c [hunk 1]
```diff
@@ -3,6 +3,19 @@
 #include "ht/ht.h"
 #include "util/macros.h"
 
+/**
+ * Calculate the bucket index for a hash and a hashtable
+ *
+ * @return The index of the hash `hash` in the hashtable `ht`
+ */
+static inline size_t
```
Comment (TheNeikos) [inline src/libreset/ht/base.c:13]: I would suggest using `R_FORCE_INLINE` here.

Comment (matthiasbeyer) [inline src/libreset/ht/base.c:13]: Hm, would like to hear @neithernut s opinion.

Comment (neithernut) [inline src/libreset/ht/base.c:13]: We might do this, _after_ the attribute header is fixed.

**Comment by neithernut** [comment]
`bucket_index` could be forward declarated, as we did with some AVL functions.
Having all the internals at the end of the file is cleaner imho.

**Comment by matthiasbeyer** [comment]
I don't want to add extra noise to the file with this, @neithernut .

**Comment by neithernut** [comment]
k, then.
ACK.

**Pull request merged**